src/jtag/drivers/picoprobe.c

```diff
@@ -477,11 +478,51 @@ static const struct swd_driver picoprobe_swd = {
 	.run = picoprobe_swd_run_queue,
 };
 
+const char *useSerialNumber = NULL;
+static COMMAND_HELPER(handle_serialnum_args, const char **serialNumber)
+{
+  if(CMD_ARGC != 1)
+  {
+    LOG_ERROR("%s: need single argument with serial number", CMD_NAME);
```

Comment:
Given that this is specific to pico_serialnum would it also be worth erroring if serialNumber doesn't have the right length?

Reply:
I wasn't sure if this may change in the future, also you get a decent error message from openocd if the serial number is not found.